### PR TITLE
Improve some schema errors that relied on debug formatting of sets

### LINKED ISSUE #832: Improve schema parsing error messages that rely on `Debug` formatting of fields
### Category

Error message improvements

### Describe the feature you'd like to request

For example, an `UndeclaredActions` error uses the debug formatting of a `HashSet`, so it prints as

```
undeclared action(s): {"Action::\"photo_action\""}
```

Notable this messages includes unnecessary curly braces, quotes around the action, and escapes the quotes on the euid. We should format these into something nicer like

```
undeclared action: `Action::"photo_action"`
```

Relevant errors are

- [ ] `UndeclaredActions`
- [ ] `UndeclaredEntityTypes`
- [ ] `UndeclaredCommonTypes`
- [ ] `ToHumanSchemaStrError::NameCollisions`

### Describe alternatives you've considered

.

### Additional context

_No response_

### Is this something that you'd be interested in working on?

- [ ] 👋 I may be able to implement this feature request
- [ ] ⚠️ This feature might incur a breaking change

### PR DESCRIPTION
**Pull request by john-h-kastner-aws**
## Issue #, if available
Fix #832.
Also update the undeclared common type error message for #766, update some tests to check error message text, and use `BTreeSet` in error structs so that error text is deterministic. 

## Checklist for requesting a review

The change in this PR is (choose one, and delete the other options):

- [ ] A breaking change requiring a major version bump to `cedar-policy` (e.g., changes to the signature of an existing API).
- [ ] A backwards-compatible change requiring a minor version bump to `cedar-policy` (e.g., addition of a new API).
- [x] A bug fix or other functionality change requiring a patch to `cedar-policy`.
- [ ] A change "invisible" to users (e.g., documentation, changes to "internal" crates like `cedar-policy-core`, `cedar-validator`, etc.)
- [ ] A change (breaking or otherwise) that only impacts unreleased or experimental code.

I confirm that this PR (choose one, and delete the other options):

- [ ] Updates the "Unreleased" section of the CHANGELOG with a description of my change (required for major/minor version bumps).
- [x] Does not update the CHANGELOG because my change does not significantly impact released code.

I confirm that [`cedar-spec`](https://github.com/cedar-policy/cedar-spec) (choose one, and delete the other options):

- [x] Does not require updates because my change does not impact the Cedar formal model or DRT infrastructure.
- [ ] Requires updates, and I have made / will make these updates myself. (Please include in your description a timeline or link to the relevant PR in `cedar-spec`, and how you have tested that your updates are correct.)
- [ ] Requires updates, but I do not plan to make them in the near future. (Make sure that your changes are hidden behind a feature flag to mark them as experimental.)
- [ ] I'm not sure how my change impacts `cedar-spec`. (Post your PR anyways, and we'll discuss in the comments.)

### CODE REVIEW

File: cedar-policy-validator/src/err.rs [hunk 1]
```diff
@@ -146,9 +146,8 @@ pub enum SchemaError {
     #[error(transparent)]
     #[diagnostic(transparent)]
     UndeclaredActions(#[from] schema_errors::UndeclaredActionsError),
-    /// This error occurs in either of the following cases (see discussion on #477):
-    ///     - undeclared common type(s) appearing in entity or context attributes
-    ///     - common type(s) (declared or not) appearing in declarations of other common types
+    /// This error when undeclared common type appear in entity or context
+    /// attributes.
```
Comment (khieta) [inline cedar-policy-validator/src/err.rs:150]:
```suggestion
    /// This error occurs when an undeclared common type appears in entity or context
    /// attributes.
```

**Review by khieta — APPROVED** [review]
Looks great. Nice updates in the tests too.

**Comment by john-h-kastner-aws** [comment]
> errors except `ToHumanSchemaStrError::NameCollisions`

This was already updated when a public copy of the enum as added to `cedar-policy`, but the old error string still existed in the validator crate

**Review by shaobo-he-aws — APPROVED** [review]
LGTM

**Pull request merged**